Formula/rust-analyzer.rb

```diff
@@ -0,0 +1,17 @@
+class RustAnalyzer < Formula
+  desc "Experimental Rust compiler front-end for IDEs"
+  homepage "https://rust-analyzer.github.io/"
+  url "https://github.com/rust-analyzer/rust-analyzer/archive/2020-04-20.tar.gz"
+  sha256 "da47d24f24205c77bd8aeba4f0c2d6b7b12f2462ceab7e19473282c9946bd69c"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked", "--root", prefix, "--path",
+           "crates/rust-analyzer", "--bin", "rust-analyzer"
+  end
+
+  test do
+    system "#{bin}/rust-analyzer", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I added more than '--version' test.